x/auth/keeper/deterministic_test.go

```diff
@@ -118,3 +124,425 @@ func pubkeyGenerator(t *rapid.T) *rapid.Generator[secp256k1.PubKey] {
+		res, err := suite.queryClient.Params(suite.ctx, &types.QueryParamsRequest{})
+
+		suite.Require().NoError(err)
+		suite.Require().NotNil(res)
+
+		if !prevRes.Equal(types.Params{}) {
+			suite.Require().Equal(res.Params, prevRes)
+		}
+
+		prevRes = res.Params
+	}
+}
+
+func (suite *DeterministicTestSuite) TestGRPCQueryParameters() {
+	rapid.Check(suite.T(), func(t *rapid.T) {
+		params := types.NewParams(
+			rapid.Uint64Min(1).Draw(t, "max-memo-characters"),
+			rapid.Uint64Min(1).Draw(t, "tx-sig-limit"),
+			rapid.Uint64Min(1).Draw(t, "tx-size-cost-per-byte"),
+			rapid.Uint64Min(1).Draw(t, "sig-verify-cost-ed25519"),
+			rapid.Uint64Min(1).Draw(t, "sig-verify-cost-Secp256k1"),
+		)
+
+		err := suite.accountKeeper.SetParams(suite.ctx, params)
+		suite.Require().NoError(err)
+
+		suite.runParamsIterations(params)
+	})
+
+	// Regression test
+	params := types.NewParams(15, 167, 100, 1, 21457)
+
+	err := suite.accountKeeper.SetParams(suite.ctx, params)
+	suite.Require().NoError(err)
+
+	suite.runParamsIterations(params)
+}
+
+func (suite *DeterministicTestSuite) runAccountInfoIterations(addr sdk.AccAddress, prevRes *types.BaseAccount) {
+	for i := 0; i < 1000; i++ {
+		res, err := suite.queryClient.AccountInfo(suite.ctx, &types.QueryAccountInfoRequest{Address: addr.String()})
+		suite.Require().NoError(err)
+		suite.Require().NotNil(res)
+		suite.Require().NotNil(res.Info)
+
+		if prevRes != nil {
+			suite.Require().Equal(res.GetInfo().Address, prevRes.Address)
+			suite.Require().True(res.GetInfo().PubKey.Equal(prevRes.PubKey))
+			suite.Require().Equal(res.GetInfo().AccountNumber, prevRes.GetAccountNumber())
+			suite.Require().Equal(res.GetInfo().Sequence, prevRes.Sequence)
+		}
+
+		prevRes = res.GetInfo()
+	}
+}
+
+func (suite *DeterministicTestSuite) TestGRPCQueryAccountInfo() {
+	rapid.Check(suite.T(), func(t *rapid.T) {
+		pub := pubkeyGenerator(t).Draw(t, "pubkey")
+		addr := sdk.AccAddress(pub.Address())
+		accNum := rapid.Uint64().Draw(t, "account-number")
+		seq := rapid.Uint64().Draw(t, "sequence")
+
+		acc1 := types.NewBaseAccount(addr, &pub, accNum, seq)
+		suite.accountKeeper.SetAccount(suite.ctx, acc1)
+
+		suite.runAccountInfoIterations(addr, acc1)
+	})
+
+	// Regression test
+	addr1 := sdk.MustAccAddressFromBech32("cosmos1j364pjm8jkxxmujj0vp2xjg0y7w8tyveuamfm6")
+	pub, err := hex.DecodeString("01090C02812F010C25200ED40E004105160196E801F70005070EA21603FF06001E")
+	suite.Require().NoError(err)
+
+	accNum := uint64(10087)
+	seq := uint64(0)
+
+	acc1 := types.NewBaseAccount(addr1, &secp256k1.PubKey{Key: pub}, accNum, seq)
+
+	suite.accountKeeper.SetAccount(suite.ctx, acc1)
+	suite.runAccountInfoIterations(addr1, acc1)
+}
+
+func (suite *DeterministicTestSuite) createAndReturnQueryClient(ak keeper.AccountKeeper) types.QueryClient {
+	queryHelper := baseapp.NewQueryServerTestHelper(suite.ctx, suite.encCfg.InterfaceRegistry)
+	types.RegisterQueryServer(queryHelper, ak)
+	return types.NewQueryClient(queryHelper)
+}
```

Comment:
why do we need this, instead of using `suite.queryClient`?

Reply:
Because in `TestGRPCQueryBech32Prefix` we are creating a new accountKeeper instance `ak` with different prefix every time, to query that prefix we need new `queryClient` every time.